src/common/http.ts

```diff
@@ -79,10 +79,51 @@ export class TurboHTTPService implements TurboHTTPServiceInterface {
     headers?: Partial<TurboSignedRequestHeaders> & Record<string, string>;
     data: Readable | Buffer | ReadableStream;
   }): Promise<T> {
-    return this.tryRequest(
-      () => this.axios.post<T>(endpoint, data, { headers, signal }),
-      allowedStatuses,
-    );
+    // Buffer and Readable → keep Axios (streams work fine there)
+    if (!(data instanceof ReadableStream)) {
+      return this.tryRequest(
+        () => this.axios.post<T>(endpoint, data, { headers, signal }),
```

Comment:
have we considered replacing axios with native `fetch`?

Reply:
I think thats fine but doesn't solve the issue in this context. fetch is also broken here, firefox and safari does not readablestreams for http agents

Reply:
we are taking advantage of native fetch in browser contexts now -- for the chrome readable stream here and for the blob in firefox/safari